### PR TITLE
Plan trip

### PR DESCRIPTION
**Pull request by petermarathas**
Check it over lemme know, haven't tested on Android but it works fine on iOS.

:bowtie:

### CODE REVIEW

**Comment by akaplo** [comment]
Pumped to see a PR!  I just pulled your branch and played around with it in my browser, and I have two comments off the bat:
1. I see that the Steps are defined and used, but I don't see them.  (see screenshot)
   screen shot 2016-03-30 at 9 39 19 pm

I thought I might've just needed to scroll down, but it won't let me :stuck_out_tongue: 
- console error states `error    TypeError: undefined is not an object (evaluating 'step['transit']['departureTime']'), http://localhost:8100/js/controllers/plan-trip-controller.js, Line: 191`
1. When I've saved a trip and I try to select it in MyBuses, it does nothing
   - console error states `can't find variable $location on MyBusesController, line 43`

File: www/js/app.js [hunk 1]
```diff
@@ -10,129 +10,133 @@ angular.module('pvta', ['ionic', 'ngCordova', 'pvta.controllers', 'angularMoment
 
 .run(function ($ionicPlatform) {
   $ionicPlatform.ready(function () {
-    // Hide the accessory bar by default (remove this to show the accessory bar above the keyboard
-    // for form inputs)
+                // Hide the accessory bar by default (remove this to show the accessory bar above the keyboard
```
Comment (akaplo) [inline www/js/app.js]: Any reason for the random indents? haha

File: www/js/app.js [hunk 2]
```diff
@@ -10,129 +10,133 @@ angular.module('pvta', ['ionic', 'ngCordova', 'pvta.controllers', 'angularMoment
...
+         * Only after the device has specifically told us it's ready
+         * may we access the global StatusBar object.
+         * **************************************************/
   document.addEventListener('deviceready', onDeviceReady, true);
   function onDeviceReady () {
     StatusBar.backgroundColorByHexString('#387ef5');
   }
 })
 
-.config(function ($stateProvider, $urlRouterProvider, $ionicFilterBarConfigProvider) {
```
Comment (akaplo) [inline www/js/app.js:32]: Curious to know the reason behind the removal of the `$ionicFilterBarConfigProvider`. Is it unnecessary?

File: www/js/app.js [hunk 3]
```diff
@@ -10,129 +10,133 @@ angular.module('pvta', ['ionic', 'ngCordova', 'pvta.controllers', 'angularMoment
...
-      views: {
-        'menuContent': {
-          templateUrl: 'templates/vehicle.html',
-          controller: 'VehicleController'
-        }
-      }
-    })
+  }
+})
+.state('app.vehicle', {
```
Comment (akaplo) [inline www/js/app.js]: This basically just undoes what's already in Master.  I feel like you haven't pulled/merged your branch with master properly?

File: www/js/controllers/my-buses-controller.js [hunk 4]
```diff
@@ -29,4 +32,16 @@ angular.module('pvta.controllers').controller('MyBusesController', function ($sc
...
   };
+
+  $scope.removeTrip = function (index) {
+    Trips.remove(index);
+    $scope.trips.splice(index, 1);
+  };
+
+  $scope.openTrip = function (index) {
+    Trips.push(index);
+    $location.path('app/plan-trip');
```
Comment (akaplo) [inline www/js/controllers/my-buses-controller.js]: Notice that you haven't imported this on line 1.

File: www/js/controllers/plan-trip-controller.js [hunk 5]
```diff
@@ -0,0 +1,283 @@
+angular.module('pvta.controllers').controller('PlanTripController', function ($scope, $location, $cordovaGeolocation, $cordovaDatePicker, $ionicPopup, $ionicScrollDelegate, Trips) {
+
```
Comment (akaplo) [inline www/js/controllers/plan-trip-controller.js:14]: In general, I'd love some comments for this file just cause it's long...but this is a lower priority.

File: www/js/controllers/plan-trip-controller.js [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+    if (!$scope.loaded) {
+      $scope.params.name = '';//Clears the name instead of 'New Trip'
+    }
+    $ionicPopup.show({
+      template: '<input type="text" ng-model="params.name">',
+      title: 'Trip Name',
+      subTitle: 'Give this trip a name.',
+      scope: $scope,
+      buttons: [
+  {text: 'Cancel',
```
Comment (akaplo) [inline www/js/controllers/plan-trip-controller.js:276]: Uh oh, indentation!

**Comment by akaplo** [comment]
That and preliminary line notes aside though, I just poured over your code and it's a fantastic addition.

**Comment by akaplo** [comment]
There's also a `.DS_STORE` file (created by OS X for...reasons) that was committed to the /www directory. It's in the gitignore so not entirely sure how it ended up there, but...it probably shouldn't be?

File: www/js/app.js (hunk 1 above)
Comment (petermarathas) [inline www/js/app.js]: Me not using vim correctly

File: www/js/app.js (hunk 2 above)
Comment (petermarathas) [inline www/js/app.js:32]: Linter told me it was unused, so I removed it, could be wrong.

**Comment by petermarathas** [comment]
The big things that were broken are now fixed, mostly symptoms of linting which changed some things from snake to camel case when they needed to be snake..

Try it now, it should work fine.

**Comment by akaplo** [comment]
It works and looks great, amazing job!

**Comment by akaplo** [comment]
I have 2 usability comments.  These can be feature additions for another release if you feel like they'd be a lot of work, but here they are. Let me know your thoughts:
1. When you're looking at the steps for a trip, clicking on one brings you to the stop that it's talking about, which is great.  When you come back to the trip, however, you're not looking at the details for that trip anymore.  This is fine, but it would be nice for there to be a way that either
   
   **a**. You can reselect your trip from a dropdown on the page so that it can pop back up, or
   
   **b**. It would remember what trip you were looking at so it'll load it back up.
   
   I'd assume **a** would be easier to implement.
2. The UI for selecting a date/time to begin your trip might be confusing for an end-user. (see screenshot below).  If we could line up those boxes **and** change the format of the Date object, those would both go a huge way in improving an already great experience.

It's completely your call as to whether you'd like to address these now vs opening issues for them and addressing them later.  I'm indifferent; merging this as is would be fine.

The _very very_ last thing is cleaning up some of that indentation and the undoing-master issues (namely [the vehicle state that has had params added since you branched off](https://github.com/umts/pvta-multiplatform/pull/110/files#diff-bca463ed8ccf5e2058dfe307a95e8501L61) before we merge.

But other than that, this is great and good to go.

**Comment by petermarathas** [comment]
Screenshot?

**Comment by petermarathas** [comment]
That last commit is a lie.

File: www/js/services.js [hunk 7]
```diff
@@ -313,6 +390,12 @@ angular.module('pvta.services', ['ngResource'])
   var windows = [];
   function addMapListener(marker, onClick){
     google.maps.event.addListener(marker, 'click', function () {
+<<<<<<< HEAD
```
Comment (akaplo) [inline www/js/services.js]: Merge conflict that hasn't been resolved

File: www/js/services.js [hunk 8]
```diff
@@ -342,6 +426,14 @@ angular.module('pvta.services', ['ngResource'])
 
   return {
     placeDesiredMarker: placeDesiredMarker,
+<<<<<<< HEAD
```
Comment (akaplo) [inline www/js/services.js]: Another merge conflict

**Comment by akaplo** [comment]
Good to go.  Good work!

**Pull request merged**